openfe/protocols/openmm_utils/multistate_analysis.py

```diff
@@ -0,0 +1,619 @@
+            # pymbar 3
+            overlap_matrix = self.analyzer.mbar.computeOverlap()
+        except AttributeError:
+            overlap_matrix = self.analyzer.mbar.compute_overlap()
+
+        return overlap_matrix
+
+    def get_exchanges(self) -> dict[str, npt.NDArray]:
+        """
+        Gather both the transition matrix (and relevant eigenvalues) between
+        replicas.
+
+        Return
+        ------
+        transition_matrix : dict[str, npt.NDArray]
+          A dictionary containing the following:
+            * ``eigenvalues``: The sorted (descending) eigenvalues of the
+              lambda state transition matrix
+            * ``matrix``: The transition matrix estimate of a replica switchin
+              from state i to state j.
+        """
+        # Get replica mixing statistics
+        mixing_stats = self.analyzer.generate_mixing_statistics()
+        transition_matrix = {'eigenvalues': mixing_stats.eigenvalues,
+                             'matrix': mixing_stats.transition_matrix}
+        return transition_matrix
+
+    @property
+    def replica_states(self):
+        """
+        Timeseries of states for each replica.
+        """
+        return self._replica_states
+
+    @property
+    def equilibration_iterations(self):
+        """
+        Number of iterations discarded as equilibration.
+        """
+        return self._equil_iters
+
+    @property
+    def production_iterations(self):
+        """
+        Number of production iterations from which energies are sampled.
+        """
+        return self._prod_iters
+
+    @property
+    def free_energy(self):
+        """
+        The free energy estimate from decorrelated unbiased samples
+        """
+        return self._free_energy
+
+    @property
+    def free_energy_error(self):
+        """
+        The MBAR estimate of the free energy estimate
+        """
+        return self._free_energy_err
+
+    @property
+    def forward_and_reverse_free_energies(self):
+        """
+        The dictionary forward and reverse analysis of the free energies
+        using the number of samples defined at class initialization
+        """
+        return self._forward_reverse
+
+    @property
+    def free_energy_overlaps(self):
+        """
+        A dictionary containing the estimated overlap matrix and corresponding
+        eigenvalues and scalars of the free energies.
+        """
+        return self._overlap_matrix
+
+    @property
+    def replica_exchange_statistics(self):
+        """
+        A dictionary containing the estimated replica exchange matrix
+        and corresponding eigenvalues.
+        """
+        if hasattr(self, '_exchange_matrix'):
+            return self._exchange_matrix
+        else:
+            errmsg = ("Exchange matrix was not generated, this is likely "
+                      f"{self.sampling_method} is not repex.")
+            raise ValueError(errmsg)
+
+    @property
+    def _unit_results_dict(self):
```

Comment:
why is this a private method which is required elsewhere?  Why are some keys "private"?  It just feels like we're trying to make this hard to use

Reply:
The private keys is because @dwhswenson asked for it and I didn't want to cycle on that conversation and avoid merging anytime this side of the year because I somehow introduced something that broke the "public facing" unit result.

Re: private method, we can make it public - I don't really care, I did it because I don't expect anything else but the units to use this.